src/Redux/Store/Expeditions/Expeditions/sideEffects/helpers.ts

```diff
@@ -187,7 +187,57 @@ export const getMageIds = ({
     return randomTreasureConfigs.reduce(
       (acc: RewardResult, _) => {
         const filteredMageIds = stillAvailableMageIds.filter(
-          mageId => ids.indexOf(mageId) === -1
+          (mageId) => ids.indexOf(mageId) === -1
+        )
+
+        const mageIdsResult = createIdList(
+          filteredMageIds,
+          ['EMPTY'],
+          undefined,
+          acc.seed
+        )
+
+        return {
+          result: [...acc.result, ...mageIdsResult.result],
+          seed: mageIdsResult.seed,
+        }
+      },
+      {
+        ...baseResult,
+        result: ids,
+      } as RewardResult // FIXME casting should technically not be necessary (but i could make it work without)
+    )
+  }
+}
+
+export const getMageIdsFromSimpleArray = ({
```

Comment:
Hm, this function seems to be almost identical to `getMageIds()` above. I think we should remove this additional function and use `getMageIds()` instead.

We could probably type our `initialBaracksConfig()` in a way, that uses our various rewards config types (e.g. `MageRewardsConfig` - that way it would be compatible without us doing much.

The only thing we then will have to do, is to write a new migration.

Reply:
I had thought about this. I wasn't sure if thats the route we wanted to go because technically its not really a reward, and the rewards config is just a slightly more complex object compared to what I have. But I also see the merit of not having to get both functions up to date going forward.

Reply:
yeah - I think an additional benefit of this would be the consistency for users writing these configs.